### PR TITLE
ISPN-14924 Retrieve roles and permissions

### PR DESCRIPTION
**Pull request by karesti**
https://issues.redhat.com/browse/ISPN-14924

### CODE REVIEW

**Comment by karesti** [comment]
@tristantarrant is there a way to know which are the pre defined roles?

**Comment by karesti** [comment]
@tristantarrant I have my answer, I'm adding the enhancement

**Review by tristantarrant — CHANGES_REQUESTED** [review]
We need a test in `RESTAuthorizationTest`

File: server/testdriver/core/src/main/java/org/infinispan/server/test/core/Containers.java [hunk 1]
```diff
@@ -36,7 +36,8 @@ public static String imageArchitecture() {
          case "aarch64":
             return "arm64";
          default:
-            throw new IllegalArgumentException("Unknown architecture");
+            return "amd64";
```
[the remark this answers is not in the document]
Comment (karesti) [reply server/testdriver/core/src/main/java/org/infinispan/server/test/core/Containers.java:39]: I opened a separate PR

**Comment by karesti** [comment]
@tristantarrant I added the tests

**Review by ryanemerson — CHANGES_REQUESTED** [review]
@karesti The following failures seem related:

```
[org.infinispan.server.security.authorization.AuthorizationKerberosIT$Rest.testRoleManipulation](https://ci.infinispan.org/job/Infinispan/job/PR-11326/5/testReport/junit/org.infinispan.server.security.authorization/AuthorizationKerberosIT$Rest/testRoleManipulation/)
[org.infinispan.server.security.authorization.AuthorizationKerberosIT$Rest.(?)](https://ci.infinispan.org/job/Infinispan/job/PR-11326/5/testReport/junit/org.infinispan.server.security.authorization/AuthorizationKerberosIT$Rest/___/)
[org.infinispan.server.security.authorization.AuthorizationLDAPIT$Rest.testRoleManipulation](https://ci.infinispan.org/job/Infinispan/job/PR-11326/5/testReport/junit/org.infinispan.server.security.authorization/AuthorizationLDAPIT$Rest/testRoleManipulation/)
[org.infinispan.server.security.authorization.AuthorizationPropertiesIT$Rest.testRoleManipulation](https://ci.infinispan.org/job/Infinispan/job/PR-11326/5/testReport/junit/org.infinispan.server.security.authorization/AuthorizationPropertiesIT$Rest/testRoleManipulation/)
```

**Comment by karesti** [comment]
@ryanemerson should be good now

**Comment by karesti** [comment]
relates to https://github.com/infinispan/infinispan-console/pull/413

**Comment by ryanemerson** [comment]
The `org.infinispan.rest.resources.AccessManagementResourceTest.testDetailedRolesList` failures look related.

**Comment by karesti** [comment]
@ryanemerson something I forgot to remove

**Comment by ryanemerson** [comment]
Thanks @karesti

**Pull request merged**